### PR TITLE
Updated the gitlab_project module with usage of other python module

### PR DESCRIPTION
**Pull request by dj-wasabi**
##### ISSUE TYPE

- Feature Pull Request
##### COMPONENT NAME

gitlab_project
##### ANSIBLE VERSION

```
ansible 2.1.2.0
  config file = /git/ansible/environment/ansible.cfg
  configured module search path = Default w/o overrides
```
##### SUMMARY

The python module used in the previous version seems not alive anymore. Suggestion from #2159 was to use other python module, one that is alive. This PR make use of this suggested module and added some new options. Other modules will be updated soon.

### CODE REVIEW

File: lib/ansible/modules/source_control/gitlab_project.py [hunk 1]
```diff
@@ -84,33 +84,61 @@
...
+    builds_enabled:
+        description:
+            - If a build creation for this project should be available or not.
+        choices: ["true", "false"]
+        default: false
+        version_added: "2.3"
+    public_builds:
+        description:
+            - If true, builds can be viewed by non-project-members.
+            - Will only work if "builds_enabled" is set to True.
```
Comment (gundalow) [inline lib/ansible/modules/source_control/gitlab_project.py]: Will only work when C(builds_enabled=True).

File: lib/ansible/modules/source_control/gitlab_project.py [hunk 2]
```diff
@@ -84,33 +84,61 @@
...
+        choices: ["true", "false"]
+        default: false
+        version_added: "2.3"
+    only_allow_merge_if_build_succeeds:
+        description:
+            - Set whether merge requests can only be merged with successful builds.
+            - Will only work if "builds_enabled" is set to True.
```
Comment (gundalow) [inline lib/ansible/modules/source_control/gitlab_project.py]: Will only work when C(builds_enabled=True).

File: lib/ansible/modules/source_control/gitlab_project.py [hunk 3]
```diff
@@ -84,33 +84,61 @@
...
+        choices: ["true", "false"]
+        default: false
+        version_added: "2.3"
+    container_registry_enabled:
+        description:
+            - Enable container registry for this project.
+            - Will only work if "builds_enabled" is set to True.
+        choices: ["true", "false"]
```
Comment (gundalow) [inline lib/ansible/modules/source_control/gitlab_project.py]: Will only work when C(builds_enabled=True).

**Review by gundalow — CHANGES_REQUESTED** [review]
needs_revision

File: lib/ansible/modules/source_control/gitlab_project.py (hunk 1 above)
Comment (dj-wasabi) [reply lib/ansible/modules/source_control/gitlab_project.py]: What the purpose of the C? (Just out of curiosity)

Comment (gundalow) [reply lib/ansible/modules/source_control/gitlab_project.py]: Code formatting. e.g.
http://docs.ansible.com/ansible/eos_command_module.html

> Configures the transport to use SSL if set to true only when the transport=eapi. If the transport argument is not eapi, this value is ignored.

**Review by gundalow — APPROVED** [review]
Looks good, thanks

**Comment by mainderay** [comment]
Tried to use this and discovered that python-gitlab parameters like namespace-id and import-url are written using an underscore and thus fail. In my case it was line 272:

`                project = self._gitlab.projects.create({'name': name, 'namespace_id': group_id})
`

should be:

`                project = self._gitlab.projects.create({'name': name, 'namespace-id': group_id})
`

I believe it may be the case for any python-gitlab parameter that uses a dash for separating words.

I hope this helps and thank you for the great module!

**Comment by dj-wasabi** [comment]
Hi @mainderay 

According to this page http://python-gitlab.readthedocs.io/en/stable/gl_objects/projects.html, it should be used with underscores.

**Comment by mainderay** [comment]
@dj-wasabi True, but the doc is incorrect--at least for the current version (0.19). I thought of using a dash because I was having trouble with the module and so I tried the CLI version of python-gitlab. In the help they show the parameter `namespace_id` with a dash. So their docs are out of date. I've let them know.

To be more specific I'm using the latest version of python-gitlab (0.19) and I kept getting a namespace error. I finally dug into the gitlab_project module code and first I removed the namespace_id from the code and it worked. Then I figured out python-gitlab 0.19 wanted namespace-id instead. That also works.

My task in Ansible looks like this so you can reproduce the issue:

```
 - name: 'Gitlab create and local creat, commit and push with .gitignore file.'
      gitlab_project:
        server_url: https://gitlab.com
        validate_certs: True
        login_token: 9YmtcDSuFDo11A6DWxB7
        name: '{{longname}}' 
        state: present
        issues_enabled: True
        wiki_enabled: True
        snippets_enabled: False
      delegate_to: localhost
```

I hope this helps. Can you reproduce the issue on your end?

-raymond

**Comment by resmo** [comment]
needs_info

**Comment by jxn** [comment]
I assume that all the gitlab modules will break with the current python lib used?  AFAIK, the next gitlab release removes the old version of the gitlab API.  Anyone know?  Any prayer of getting this merged?

**Comment by gpchelkin** [comment]
Any ETA for this?

**Comment by jxn** [comment]
@gpchelkin I don't think this will be done as this issue is closed and it looks like @dj-wasabi has withdrawn the PR. My guess is this will only be done if someone else decides to do the work, as Ansible / RedHat don't seem to be the ones maintaining the gitlab modules.

**Comment by resmo** [comment]
I am willing to work on it but ETA Ansible ~2.7

**Comment by dj-wasabi** [comment]
@jxn @gpchelkin I'll try another run with some PR's somewhere next few weeks.

**Pull request closed without merging**